File: backend/src/main/java/com/example/backend/entities/LuaGa.java
package com.example.backend.entities;

import java.time.LocalDate;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.Data;

@Entity
@Table(name = "LUA_GA")
@Data
public class LuaGa {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String ten_lua;
    private String giong_ga;
    private int so_luong_ban_dau;
    private int so_luong_hien_tai;
    private LocalDate ngay_nhap;
    private LocalDate ngay_xuat_du_kien;
    private String trang_thai;
    private int loi_nhuan;
    @JsonProperty("user_id")
    @Column(name = "user_id")
    private String userId;

    @OneToMany(mappedBy = "luaGa", cascade = CascadeType.ALL)
    @JsonIgnore
    private List<LichTiemThucTe> lichTiemThucTes;

    @OneToMany(mappedBy = "luaGa", cascade = CascadeType.ALL)
    @JsonIgnore
    private List<LichSuPhatTrien> lichSuPhatTriens;

    @OneToMany(mappedBy = "luaGa", cascade = CascadeType.ALL)
    @JsonIgnore
    private List<ChiPhiChanNuoi> chiPhiChanNuois;

    @OneToMany(mappedBy = "luaGa", cascade = CascadeType.ALL)
    @JsonIgnore
    private List<ThongBao> thongBaos;

    @OneToMany(mappedBy = "luaGa", cascade = CascadeType.ALL)
    @JsonIgnore
    private List<CamBienHomNay> camBienHomNays;
}

File: smart_chicken_farming/lib/model/LuaGa.dart
class LuaGa {
  final int? id;
  final String? giongGa;
  final String? tenLua;
  final DateTime ngayNhap;
  final DateTime ngayXuatDuKien;
  final int soLuongBanDau;
  int? soLuongHienTai;
  String trangThai;
  int? loiNhuan;
  final String userId;

  LuaGa({
    this.id,
    this.giongGa,
    this.tenLua,
    required this.ngayNhap,
    required this.ngayXuatDuKien,
    required this.soLuongBanDau,
    this.soLuongHienTai,
    required this.trangThai,
    this.loiNhuan,
    required this.userId,
  });

  factory LuaGa.fromJson(Map<String, dynamic> json) {
    return LuaGa(
      id: json['id'] as int,
      giongGa: json['giong_ga'] as String?,
      tenLua: json['ten_lua'] as String?,
      ngayNhap: DateTime.parse(json['ngay_nhap'] as String),
      ngayXuatDuKien: DateTime.parse(json['ngay_xuat_du_kien'] as String),
      soLuongBanDau: json['so_luong_ban_dau'] as int,
      soLuongHienTai: json['so_luong_hien_tai'] as int?,
      trangThai: json['trang_thai'] as String,
      loiNhuan: json['loi_nhuan'] as int?,
      userId: json['user_id'] as String,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'giong_ga': giongGa,
      'ten_lua': tenLua,
      'ngay_nhap': ngayNhap.toIso8601String(),
      'ngay_xuat_du_kien': ngayXuatDuKien.toIso8601String(),
      'so_luong_ban_dau': soLuongBanDau,
      'so_luong_hien_tai': soLuongHienTai,
      'trang_thai': trangThai,
      'loi_nhuan': loiNhuan,
      'user_id':userId,
    };
  }
}